state/src/replicationv2/raft.rs

```diff
@@ -178,6 +186,32 @@ impl RaftClient {
         self.raft.shutdown().await.map_err(err_str!(ReplicationV2Error::RaftTeardown))
     }
 
+    // -------------------
+    // | Wait Conditions |
+    // -------------------
+
+    /// Await the promotion of the local peer to a voter
+    pub async fn await_promotion(&self) -> Result<(), ReplicationV2Error> {
+        let timeout = Duration::from_millis(DEFAULT_PROMOTION_TIMEOUT_MS);
+        self.raft
+            .wait(Some(timeout))
```

Comment:
that is so neat

Reply:
yeah definitely a really neat abstraction to learn from